skimage/filters/thresholding.py

```diff
@@ -134,6 +135,59 @@ def wrapper(im):
                     methods=methods, verbose=verbose)
 
 
+def try_all_threshold_dict(image):
```

Comment:
I don't find this name particularly elegant. Ideally, I would prefer `try_all_threshold` -> dict and `try_all_threshold_plotted` -> mpl. But this would require a deprecation cycle...

Reply:
Agreed; have `try_all_threshold` always return results, and then add a `visualize` flag?

Reply:
@stefanv that makes it harder/impossible to return the figure and axes as done currently. At any rate, if long term we want to remove the mpl dependency for analysis, we could have `try_all_threshold` return a dict, and then have a separate function in e.g. the viewer (or whatever we call the skimage-mpl package) to visualise that dictionary.

Either way, yes, a deprecation is going to be needed. =\